test/org/sagebionetworks/bridge/dynamodb/DynamoUserConsentDaoMockTest.java

```diff
@@ -0,0 +1,300 @@
+    DynamoStudyConsent1 studyConsent;
+    
+    DynamoDBMapper mapper2;
+    
+    DynamoDBMapper mapper3;
+    
+    @Before
+    public void before() {
+        studyConsent = new DynamoStudyConsent1();
+        studyConsent.setActive(true);
+        studyConsent.setCreatedOn(DateTime.now().getMillis());
+        studyConsent.setStudyKey("test-study");
+        studyConsent.setStoragePath("test-study.12341234123");
+        
+        mapper2 = mock(DynamoDBMapper.class);
+        mapper3 = mock(DynamoDBMapper.class);
+        
+        createQueryResults(null, null);
+        
+        userConsentDao = new DynamoUserConsentDao();
+        userConsentDao.setDdbMapper2(mapper2);
+        userConsentDao.setDdbMapper3(mapper3);
+    }
+    
+    @SuppressWarnings("unchecked")
+    private void createQueryResults(DynamoUserConsent2 consent2, DynamoUserConsent3 consent3) {
+        PaginatedQueryList<DynamoUserConsent3> pqList = mock(PaginatedQueryList.class);
+        if (consent2 == null) {
+            when(pqList.isEmpty()).thenReturn(true);
+        } else {
+            when(pqList.isEmpty()).thenReturn(false);
+            when(pqList.get(0)).thenReturn(consent3);
+            when(pqList.stream()).thenReturn(Stream.of(consent3));
+        }
+        doReturn(consent2).when(mapper2).load(any());
+        doReturn(pqList).when(mapper3).query(any(), any());
+    }
+    
+    private void createQueryWithResults() {
+        DynamoUserConsent2 existingConsent2 = new DynamoUserConsent2(HEALTH_CODE, STUDY_IDENTIFIER.getIdentifier());
+        existingConsent2.setConsentCreatedOn(studyConsent.getCreatedOn());
+        existingConsent2.setSignedOn(DateTime.now().getMillis());
+
+        DynamoUserConsent3 existingConsent3 = new DynamoUserConsent3(HEALTH_CODE, STUDY_IDENTIFIER.getIdentifier());
+        existingConsent3.setConsentCreatedOn(studyConsent.getCreatedOn());
+        existingConsent3.setSignedOn(DateTime.now().getMillis());
+        
+        createQueryResults(existingConsent2, existingConsent3);
+    }
+    
+    @Test
+    public void giveConsent() {
+        userConsentDao.giveConsent(HEALTH_CODE, studyConsent);
+        
+        ArgumentCaptor<DynamoUserConsent3> argCaptor3 = ArgumentCaptor.forClass(DynamoUserConsent3.class);
+        
+        verify(mapper3).query(any(), any()); // queries and saves, doesn't speak to table2
+        verify(mapper2).load(any());
+        verify(mapper3).save(argCaptor3.capture());
+        verifyNoMoreInteractions(mapper2);
+        verifyNoMoreInteractions(mapper3);
+        
+        DynamoUserConsent3 cons3 = argCaptor3.getValue();
+        
+        assertTrue(cons3.getConsentCreatedOn() > 0);
+        assertEquals((Long)cons3.getSignedOn(), (Long)cons3.getSignedOn());
+        assertEquals(cons3.getHealthCode(), cons3.getHealthCode());
+        assertEquals(cons3.getHealthCodeStudy(), cons3.getHealthCodeStudy());
+        assertEquals(cons3.getStudyIdentifier(), cons3.getStudyIdentifier());
```

Comment:
These tests don't test anything--they just compare that a value is equal to itself.